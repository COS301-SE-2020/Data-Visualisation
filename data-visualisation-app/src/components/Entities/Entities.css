

#selectorDiv{
    background-color: blueviolet;
}
#selectorDiv :hover{
    cursor:pointer;
}

#button-explore-dataPage{
    margin: 0;
    position: relative;
    top: 50%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
}




