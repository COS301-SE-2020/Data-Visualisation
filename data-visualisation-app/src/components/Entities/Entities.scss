
@import '../../globals/globals.scss';

@mixin button__explore--backgroundColor {
    background: rgb(117,48,154);
    background: -moz-linear-gradient(97deg, rgba(117,48,154,1) 0%, rgba(89,163,224,1) 100%);
    background: -webkit-linear-gradient(97deg, rgba(117,48,154,1) 0%, rgba(89,163,224,1) 100%);
    background: linear-gradient(97deg, rgba(117,48,154,1) 0%, rgba(89,163,224,1) 100%);
    filter: progid:DXImageTransform.Microsoft.gradient(startColorstr="#75309a",endColorstr="#59a3e0",GradientType=1);
}


.ant-checkbox-group-item {
    display: inline-block;
}

#selectorDiv{
    background-color: blueviolet;
}

#selectorDiv :hover{
    cursor:pointer;
}


#titelCard{
    text-align: center;
    min-height: 80px;
    max-height: 80px;
    overflow: hidden;
    text-overflow: ellipsis;
}

.ant-card-body{
   
}

#button-explore-dataPage{
    @include button__explore--backgroundColor;
    -webkit-box-shadow: 0px 17px 18px -12px rgba(101,112,192,1);
    -moz-box-shadow: 0px 17px 18px -12px rgba(101,112,192,1);
    box-shadow: 0px 17px 18px -12px rgba(101,112,192,1);

    margin: 0;
    top: 93%;
    left: 50%;
    -ms-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    position: fixed;
    border: none;
}
#button-explore-dataPage:hover{
    height: 5vh;
}




