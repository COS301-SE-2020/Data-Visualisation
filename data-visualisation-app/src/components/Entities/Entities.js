/**
 *   @file Entities.js
 *   Project: Data Visualisation Generator
 *   Copyright: Open Source
 *   Organisation: Doofenshmirtz Evil Incorporated
 *
 *   Update History:
 *   Date        Author              Changes
 *   -------------------------------------------------------
 *   15/7/2020   Byron Tominson      Original
 *
 *   Test Cases: data-visualisation-app/src/tests/Entities.test.js
 *
 *   Functional Description:
 *   Displays entities for the user to choose from.
 *
 *   Error Messages: "Error"
 *   Assumptions: None
 *   Constraints: None
 */

import './Entities.css';
import React, { Fragment } from 'react';
import { List, Avatar, Button, Skeleton, Form, Checkbox } from 'antd';
import {CompassOutlined} from '@ant-design/icons';
import request from '../../globals/requests';
import * as constants from '../../globals/constants';

const count = 20;
const fakeDataUrl = `https://randomuser.me/api/?results=${count}&inc=name,gender,email,nat&noinfo`;



class Entities extends React.Component {

  onFinish = values => {
    console.log('Received values of form: ', values);
    this.next();
  };

  state = {
    initLoading: true,
    loading: false,
    data: [],
    list: [],
  };


  onChange = (item) => {
    console.log(item.name.last);
  };


  next = () => {
    this.props.setStage('suggestions');
  };


  
  /**
    * invoked immediately after a component is mounted (inserted into the tree).
  */
  componentDidMount() {
    console.log(request.user.isLoggedIn);
      this.getData(res => {
        this.setState({
          initLoading: false,
          data: request.user.entities,
          list: request.user.entities,
        });
      });
  }

  /**
    * Funnction uses the dataSources to update the entites list and tempEntities list for setting sourcesAndEntities list
    * which is used for making requests.
  */
  getData = callback => {
    //call on all data soureces
    request.user.dataSources.map((source) => {
      request.entities.list(source.sourceurl, function(result) {
          if (result === constants.RESPONSE_CODES.SUCCESS) {
            request.user.sourcesAndEntities = request.user.sourcesAndEntities.concat({item : [source.sourceurl ,request.user.tempEntities] } );
            console.log(request.user.sourcesAndEntities);
            callback(request.user.entities);
          }
        });
    });
  };

  render() {
    const { initLoading, loading, list } = this.state;
    const loadMore =
      !initLoading && !loading ? (
        <div
          style={{
            textAlign: 'center',
            marginTop: 12,
            height: 32,
            lineHeight: '32px',
          }}
        >
        </div>
      ) : null;
   

    return (
      
     <div>
        <Form
            name='entitiesForm'
            onFinish={this.onFinish}
        >
          <List
            className='entitesList'
            loading={initLoading}
            itemLayout='horizontal'
            loadMore={loadMore}
            dataSource={list}
            renderItem={item => (
            
              <Fragment>
                  {/*<div id = 'selectorDiv' onClick = {() => {this.onChange(item)}}>*/}
                    <List.Item
                      key={1}
                      actions={
                        [ 
                        <Form.Item name = {item} valuePropName = 'checked'>
                            <Checkbox defaultChecked = {false}></Checkbox>
                        </Form.Item>
                        ]
                      }>
                      <Skeleton avatar title={false} loading={item.loading} active>
                        <List.Item.Meta
                          avatar={
                            <Avatar src='https://15f76u3xxy662wdat72j3l53-wpengine.netdna-ssl.com/wp-content/uploads/2018/03/OData-connector-e1530608193386.png' />
                          }
                          title={<a href="https://ant.design">{item}</a>}
                          description='Ant Design, a design language for background applications, is refined by Ant UED Team'
                        />
                        <div></div>
                      </Skeleton>
                    </List.Item>
                    {/*</div>*/}
              </Fragment>
            )}
          />
          <Form.Item>
            <Button id = 'button-explore-dataPage' type="primary" htmlType="submit" shape = 'round' icon={<CompassOutlined />}>
              Generate Suggestions
            </Button>
          </Form.Item>
        </Form>
      </div>
    );
  }
}

export default Entities;
