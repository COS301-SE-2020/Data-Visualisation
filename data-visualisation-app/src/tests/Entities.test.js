import React from 'react';
import Entities from '../components/Entities/Entities';
import renderer from 'react-test-renderer';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

describe('DataConnection component', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Entities/>);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});