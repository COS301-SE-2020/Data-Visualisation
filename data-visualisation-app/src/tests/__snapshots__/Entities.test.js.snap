// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataConnection component renders correctly 1`] = `
<div>
  <form
    className="ant-form ant-form-horizontal"
    id="entitiesForm"
    onSubmit={[Function]}
  >
    <div
      className="ant-card titleCard"
      style={
        Object {
          "backgroundColor": "transparent",
          "margin": "0 auto",
          "marginTop": "20px",
          "width": "80%",
        }
      }
    >
      <div
        className="ant-card-head"
        style={
          Object {
            "border": 0,
            "textAlign": "center",
          }
        }
      >
        <div
          className="ant-card-head-wrapper"
        >
          <div
            className="ant-card-head-title"
          >
            Select Entities From Your Datasources
          </div>
        </div>
      </div>
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <label
          className="ant-checkbox-wrapper"
        >
          <span
            className="ant-checkbox"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-checkbox-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="checkbox"
            />
            <span
              className="ant-checkbox-inner"
            />
          </span>
          <span>
            Check All
          </span>
        </label>
        <div
          className="ant-list entitesList ant-list-split ant-list-loading"
        >
          <div
            className="ant-spin-nested-loading"
          >
            <div>
              <div
                className="ant-spin ant-spin-spinning"
              >
                <span
                  className="ant-spin-dot ant-spin-dot-spin"
                >
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                  <i
                    className="ant-spin-dot-item"
                  />
                </span>
              </div>
            </div>
            <div
              className="ant-spin-container ant-spin-blur"
            >
              <div
                style={
                  Object {
                    "minHeight": 53,
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={Object {}}
    >
      <div
        className="ant-col ant-form-item-control"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="blob-div"
            >
              <button
                className="preblob-btn"
                icon={<ForwardRef(CompassOutlined) />}
                id="loggedOut_generateSuggestions"
                shape="round"
                size="large"
                type="submit"
              >
                Generate Suggestions
                <span
                  className="preblob-btn__inner"
                >
                  <span
                    className="preblob-btn__blobs"
                  >
                    <span
                      className="preblob-btn__blob"
                    />
                    <span
                      className="preblob-btn__blob"
                    />
                    <span
                      className="preblob-btn__blob"
                    />
                    <span
                      className="preblob-btn__blob"
                    />
                  </span>
                </span>
              </button>
              <br />
              <svg
                version="1.1"
                xmlns="http://www.w3.org/2000/svg"
              >
                <defs>
                  <filter
                    id="goo"
                  >
                    <feGaussianBlur
                      in="SourceGraphic"
                      result="blur"
                      stdDeviation="10"
                    />
                    <feColorMatrix
                      in="blur"
                      mode="matrix"
                      result="goo"
                      values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 21 -7"
                    />
                    <feBlend
                      in="SourceGraphic"
                      in2="goo"
                      result="mix"
                    />
                  </filter>
                </defs>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
